### ISSUE TITLE
Door not closing

### ISSUE THREAD

**Issue by jcbigears** [hjdhjd/homebridge-myq2#23]
Hi there. 
Having an issue with the door not closing. It’s perhaps a polling issue that I need to adjust - I'll accept the blame entirely! The door opens under command of either HomeKit or within Homebridge, but doesn’t close. I can only seem to make it close manually. Can you help? I’ve attached my logs from today. 
[logs.txt](https://github.com/hjdhjd/homebridge-myq2/files/4688268/logs.txt)

Have tried the other Chamberlain apps but keep coming back to this one. Keep up the good work!

**Comment by hjdhjd** [comment]
@jcbigears Thanks! So...it's a polling issue from what I can see in the logs. I've had this problem myself. The challenge is that this is a polling API, not a push / notification API. So there's no way to ask MyQ to **tell** us that things are opening or closing.

The workaround that all the unofficial MyQ plugins use is to poll at regular intervals and hope to catch the open and close events.

The problem is that it's possible to have an open and a close event in between polling intervals and you would miss it.

The defaults try to account for it - `longPoll` is 15 seconds. Meaning we check every 15 seconds for an event. So every 15 seconds it checks to see if something is happening (opening / closing).

`shortPoll` is the attempt to more aggressively look for events once we do detect something is going on. Meaning now I've detected that the garage door is opening, I want to keep checking to see if it opens fully. That defaults to 5 seconds and seems reasonable for most use cases.

`shortPollDuration` is how long we continue to poll for events aggressively. That is 10 minutes by default.

The "resolution" if you will, for events, in practice is pretty good. That said, it's possible to entirely miss events in two scenarios:

- Say your garage door is closed. You start opening the garage door right at the beginning of that 15 second `longPoll` interval. The open completes before we next check for events. That seems to be what is happening in your logs.

Normally, you go from garage closed -> opening -> open. If `longPoll` doesn't trigger in time to catch the opening event, it will be missed and what MyQ2 will see when it does poll is that the garage has now gone from closed to open, without an intervening opening event.

- Say your garage door is closed. You start opening the garage door and we do in fact catch the opening event. Now we shift to `shortPoll` duration polling to keep checking for event changes with a higher "resolution" so we can hopefully catch all of the events we need to see.

Again, It's possible to miss events if they complete in between `shortPoll` intervals.

I haven't seen much of this second scenario since I tuned the defaults a couple of years ago, and from what I'm seeing in your logs, it's `longPoll` you want to adjust. Your garage door seems to just open too fast. :smile:

A word of caution. I would be careful about lowering this number to anything lower than 10 seconds for an extended period of time. There are lots of reports of Chamberlain / Liftmaster blacklisting controllers who overtax their servers.

Of course the right fix for this is to give us a proper, open, notification API, but I digress. :smile:

TL;DR: adjust `longPoll` down. You may want to measure how long it takes for your garage door to actually open / close and use that to help you calibrate the `longPoll` interval. Worst case...you can slow down how quickly your garage door opens and closes to ensure it gets caught by `longPoll`. Just sayin'...

**Comment by whreams** [comment]
Is it normal to see the below?  The timestamps are different and I only have one plugin, one instance and one door...

![image](https://user-images.githubusercontent.com/64332758/86137611-17d57980-bab3-11ea-9c1c-230bf47bc094.png)
